### PR TITLE
wall mounts now have consistent layering

### PR DESCRIPTION
**Pull request by ghost**
## Changelog
:cl:
add: Added new things
add: Added more things
del: Removed old things
qol: made something easier to use
balance: rebalanced something
fix: fixed a few things
soundadd: added a new sound thingy
sounddel: removed an old sound thingy
imageadd: added some icons and images
imagedel: deleted some icons and images
spellcheck: fixed a few typos
code: changed some code
refactor: refactored some code
config: changed some config setting
admin: messed with admin stuff
server: something server ops should know
/:cl:

### CODE REVIEW

File: code/datums/elements/wall_mount.dm [hunk 1]
```diff
@@ -24,3 +23,22 @@
...
+	RegisterSignal(target, COMSIG_ATOM_DIR_CHANGE, .proc/on_dir_changed)
+	on_dir_changed(real_target, real_target.dir, real_target.dir)
+
+/datum/element/wall_mount/Detach(datum/source, ...)
+	. = ..()
+	UnregisterSignal(source, COMSIG_ATOM_DIR_CHANGE)
+
+
+/datum/element/wall_mount/proc/on_dir_changed(datum/target, olddir, newdir)
+	var/atom/movable/real_target = target
```
Comment (tralezab) [inline code/datums/elements/wall_mount.dm:35]: just make `datum/target` typecasted as a movable atom in the first place

File: code/datums/elements/wall_mount.dm [hunk 2]
```diff
@@ -24,3 +23,22 @@
 				real_target.pixel_y = 16
 
 	//target.RegisterSignal(target_to_listen_to, COMSIG_TURF_CHANGE, )
+	RegisterSignal(target, COMSIG_ATOM_DIR_CHANGE, .proc/on_dir_changed)
+	on_dir_changed(real_target, real_target.dir, real_target.dir)
```
Comment (tralezab) [inline code/datums/elements/wall_mount.dm:27]: i'm lovin' it

File: code/datums/elements/wall_mount.dm [hunk 3]
```diff
@@ -24,3 +23,22 @@
 				real_target.pixel_y = 16
 
 	//target.RegisterSignal(target_to_listen_to, COMSIG_TURF_CHANGE, )
```
Comment (tralezab) [inline code/datums/elements/wall_mount.dm:25]: remove pseudocode why not

**Review by tralezab — COMMENTED** [review]
more small buggery

File: code/datums/elements/wall_mount.dm (hunk 3 above)
Comment (ghost) [reply code/datums/elements/wall_mount.dm:25]: n o

**Pull request merged**